src/RMPI/mpidata.jl

```diff
@@ -127,17 +127,61 @@ function Rimu.DictVectors.walkernumber(md::MPIData)
     return MPI.Allreduce(walkernumber(md.data), +, md.comm)
 end
 
+function Base.:*(lop, md::MPIData)
+    T = promote_type(eltype(lop),valtype(md))
+    P = Pair{keytype(md),T}
+    buffers = Vector{P}[P[] for _ in 1:mpi_size()]
```

Comment:
I believe this is equivalent to the simpler code
```suggestion
    buffers = [P[] for _ in 1:mpi_size()]
```

Reply:
It doesn't infer if `Vector{P}` is missing. It thinks `mpi_size()` can be zero, in which case it would have resulted in a `Vector{Any}`. This is a bit weird because something `[i for i in 1:-1]` infers fine.

Reply:
Thanks!